content/en/docs/concepts/services-networking/ingress.md

```diff
@@ -13,48 +13,90 @@ weight: 40
-  [ Services ]
-```
-
-An Ingress is a collection of rules that allow inbound connections to reach the cluster services.
-
-```
-    internet
-        |
    [ Ingress ]
    --|-----|--
    [ Services ]
 ```
 
-It can be configured to give services externally-reachable URLs, load balance traffic, terminate SSL, offer name based virtual hosting, and more. Users request ingress by POSTing the Ingress resource to the API server. An [Ingress controller](#ingress-controllers) is responsible for fulfilling the Ingress, usually with a loadbalancer, though it may also configure your edge router or additional frontends to help handle the traffic in an HA manner.
+An ingress can be configured to give services externally-reachable URLs, load balance traffic, terminate SSL, and offer name based virtual hosting. An [ingress controller](#ingress-controllers) is responsible for fulfilling the ingress, usually with a loadbalancer, though it may also configure your edge router or additional frontends to help handle the traffic.
+
+An ingress does not expose arbitrary ports or protocols. Exposing services other than HTTP and HTTPS to the internet typically
+uses a service of type [Service.Type=NodePort](/docs/concepts/services-networking/service/#nodeport) or
+[Service.Type=LoadBalancer](/docs/concepts/services-networking/service/#loadbalancer).
 
 ## Prerequisites
 
-Before you start using the Ingress resource, there are a few things you should understand. The Ingress is a beta resource, not available in any Kubernetes release prior to 1.1. You need an Ingress controller to satisfy an Ingress, simply creating the resource will have no effect.
+{{< feature-state for_k8s_version="v1.1" state="beta" >}}
+
+Before you start using an ingress, there are a few things you should understand. The ingress is a beta resource. You will need an ingress controller to satisfy an ingress, simply creating the resource will have no effect.
+
+GCE/Google Kubernetes Engine deploys an [ingress controller](#ingress-controllers) on the master. Review the
+[beta limitations](https://github.com/kubernetes/ingress-gce/blob/master/BETA_LIMITATIONS.md#glbc-beta-limitations)
+of this controller if you are using GCE/GKE.
+
+In environments other than GCE/Google Kubernetes Engine, you may need to
+[deploy an ingress controller](https://kubernetes.github.io/ingress-nginx/deploy/). There are a number of
+[ingress controller](#ingress-controllers) you may choose from.
+
+## Ingress controllers
+
+In order for the ingress resource to work, the cluster must have an ingress controller running. This is unlike other types of controllers, which run as part of the `kube-controller-manager` binary, and are typically started automatically with a cluster. Choose the ingress controller implementation that best fits your cluster.
+
+* Kubernetes as a project currently supports and maintains [GCE](https://git.k8s.io/ingress-gce/README.md) and
+  [nginx](https://git.k8s.io/ingress-nginx/README.md) controllers.
```

Comment:
Karen's point below about repeating this line is a good one -- context makes the later one more useful, I think. Or else move this one earlier, since the previous section talks about GCE and nginx too.